### PR TITLE
Upgrade to Cairo 0.10.0 

### PR DESCRIPTION
**Pull request by hsouf**
Migrating the code base to Cairo 0.10.0

### CODE REVIEW

File: README.md
Comment (achab) [inline README.md:158]: you should mention the full version: `3.9.0`

Comment (hsouf) [reply README.md:158]: ack

File: hardhat.config.ts
Comment (achab) [inline hardhat.config.ts:22]: nice! it means we should add `ETHERSCAN_API_KEY` to github secrets, right?

File: package.json
Comment (achab) [inline package.json:16]: why not keeping the previous command as well?

File: package.json
Comment (achab) [inline package.json:54]: we don't use `devDepencencies` since the dev/prod relationship in not handled the same way for crypto. let's move it to `dependencies` instead

File: pyproject.toml
Comment (achab) [inline pyproject.toml:8]: use a fixed version of python here, 3.9.0 for instance

File: scripts/allowlistedTokens.ts
Comment (achab) [inline scripts/allowlistedTokens.ts:4]: typo: symbol should be `saUSDC`

File: scripts/deployBridge.ts
Comment (achab) [inline scripts/deployBridge.ts:16]: tiny: usually we use camelCase for typescript (so proxyAdmin and maxFee below)

**Review by achab — APPROVED** [review]
Thanks a lot for this PR @hsouf! Really useful changes to deployment scripts! Let's wait for Peckshield PR to be merged before merging this one.

File: hardhat.config.ts
Comment (hsouf) [reply hardhat.config.ts:22]: already added 👍🏻

File: .github/workflows/e2e-tests.yml
Comment (achab) [inline .github/workflows/e2e-tests.yml:12]: why do we add it? it's not used by the file

File: hardhat.config.ts
Comment (achab) [inline hardhat.config.ts:56]: we can instead set a different value for `STARKNET_NETWORK` in the CI

File: hardhat.config.ts
Comment (achab) [inline hardhat.config.ts:56]: I feel that this value is equal to `"l2_testnet"` locally and in the CI. When should `L2_NETWORK` be different than `"l2_testnet"`?

Comment (hsouf) [reply hardhat.config.ts:56]: I use it for deployment.

File: scripts/deployTokens.ts
Comment (hsouf) [inline scripts/deployTokens.ts:22]: typo: `token's`

File: scripts/deployBridge.ts
Comment (hsouf) [inline scripts/deployBridge.ts:109]: should be added as a param to doc.

File: scripts/deployBridge.ts
Comment (hsouf) [inline scripts/deployBridge.ts:26]: nitpick: l2GovRelay to be added as param to doc

File: scripts/deploy.ts
Comment (hsouf) [inline scripts/deploy.ts:114]: let's remove this comment

File: scripts/deploy.ts
Comment (hsouf) [inline scripts/deploy.ts:78]: let's remove this comment

File: scripts/deployBridge.ts
Comment (achab) [reply scripts/deployBridge.ts:26]: good catch!

File: scripts/deployBridge.ts
Comment (achab) [reply scripts/deployBridge.ts:109]: fixed in next commit

**Pull request merged**
